libs/payments/stripe/src/lib/util/stripeInvoiceToFirstInvoicePreviewDTO.spec.ts

```diff
@@ -2,33 +2,49 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { faker } from '@faker-js/faker';
-
 import { stripeInvoiceToFirstInvoicePreviewDTO } from './stripeInvoiceToFirstInvoicePreviewDTO';
 import { StripeResponseFactory } from '../factories/api-list.factory';
-import { StripeDiscountFactory } from '../factories/discount.factory';
-import { StripeTaxRateFactory } from '../factories/tax-rate.factory';
 import { StripeUpcomingInvoiceFactory } from '../factories/upcoming-invoice.factory';
+import { StripeTotalDiscountAmountsFactory } from '../factories/total-discount-amounts.factory';
+import { StripeTotalTaxAmountsFactory } from '../factories/total-tax-amounts.factory';
 
 describe('stripeInvoiceToFirstInvoicePreviewDTO', () => {
   it('formats invoice', () => {
     const mockUpcomingInvoice = StripeResponseFactory(
       StripeUpcomingInvoiceFactory({
-        discount: StripeDiscountFactory(),
+        total_discount_amounts: [StripeTotalDiscountAmountsFactory()],
+        total_tax_amounts: [StripeTotalTaxAmountsFactory()],
```

Comment:
If you assign these arrays to a variable, you can cleanup the code a little below, and won't need to perform a truthy check on `mockUpcomingInvoice.total_discount_amounts`. 

Something like this, using `total_discount_amounts` as an example.

```
const mockDiscountAmount = StripeTotalDiscountAmountsFactory();
...
StripeUpcomingInvoiceFactory({
  total_discount_amounts: [mockDiscountAmount],
  ...
});
...
expect(result).toEqual({
  ...
  discountAmount: mockDiscountAmount.amount,
})
```